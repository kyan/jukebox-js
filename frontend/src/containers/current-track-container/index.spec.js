import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import GoogleAuthContext from 'contexts/google'
import configureMockStore from 'redux-mock-store'
import CurrentTrackContainer from './index'

describe('CurrentTrackContainer', () => {
  let wrapper

  describe('render the connected app', () => {
    const store = configureMockStore()({
      timer: { duration: 10000, position: 8000, remaining: 700 }
    })
    const mockGoogle = {
      googleUser: { googleId: 'duncan123' }
    }

    it('renders something even when no track is available', () => {
      wrapper = mount(
        <Provider store={store}>
          <GoogleAuthContext.Provider value={mockGoogle}>
            <CurrentTrackContainer />
          </GoogleAuthContext.Provider>
        </Provider>
      )
      expect(wrapper).toMatchSnapshot()
      wrapper.find('CurrentTrack').prop('onVote')('uri123', 99)
      expect(store.getActions()).toEqual([{
        type: 'actionVote',
        key: 'vote::castVote',
        params: {
          uri: 'uri123',
          vote: 99
        }
      }])
    })
  })
})
