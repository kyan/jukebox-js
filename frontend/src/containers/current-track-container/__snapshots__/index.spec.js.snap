// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentTrackContainer render just the app without redux renders as expected 1`] = `
<CurrentTrack
  image="image"
  progress={50}
  track={
    Object {
      "track": "track",
    }
  }
/>
`;

exports[`CurrentTrackContainer render the connected app renders as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Component)>
    <Component
      dispatch={[Function]}
      image={null}
      progress={0}
    >
      <CurrentTrack
        image={null}
        progress={0}
      />
    </Component>
  </Connect(Component)>
</Provider>
`;
